### PR TITLE
task_DEV-2 calculator

### PR DESCRIPTION
**Pull request by VladislavZablocki**
Calculator which supports next operator "*","/","+","-","()" and exponential form of number

### CODE REVIEW

File: Task_DEV-2/Converter.cs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            return changedString;
+        }
+
+        // return priority of operatoin
+        private int GetPriority(string expressionOperator)
+        {
+            switch (expressionOperator)
+            {
+                case "*":
+                    return 2;
```
Comment (IgorKheidorov) [inline Task_DEV-2/Converter.cs]: Magic numbers, use enums

File: Task_DEV-2/Program.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                inputString = Console.ReadLine();
+            }
+            else
+            {
+                foreach (var item in args)
+                {
+                    inputString = String.Concat(inputString, item);
+                }
+            }
+            Checker checker = new Checker();
```
Comment (IgorKheidorov) [inline Task_DEV-2/Program.cs:29]: Where is calculator???? I see only string checker

File: Task_DEV-2/Checker.cs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                }
+            }
+            if (error)
+            {
+                Console.WriteLine("Error! Check your input data!");
+                Console.ReadKey();
+            }
+            else
+            {
+                Calculator calculator = new Calculator(inputString);
```
Comment (IgorKheidorov) [inline Task_DEV-2/Checker.cs]: This is not a proper place to call calculator, CheakInputString is for Check!!!! not to calculate

File: Task_DEV-2/Calculator.cs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                    return stack;
+                default:
+                    return stack;
+            }
+
+        }
+
+        // method which gets string in postfix format and do all operation
+        // and return result
+        public double Calculate()
```
Comment (IgorKheidorov) [inline Task_DEV-2/Calculator.cs:75]: Can't understand nothing, what is stack? what is Infix and postfix conversion, what  for?

File: Task_DEV-2/Calculator.cs [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                            stack[stackCount - i] = 0;
+                            break;
+                        }
+                    }
+                    isNumber[stackCount] = true;
+                    break;
+                }
+
+            }
+            switch (operation)
```
Comment (IgorKheidorov) [inline Task_DEV-2/Calculator.cs:53]: Switch on strings is not a good idea, use enums

File: Task_DEV-2/Checker.cs [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+                {
+                    closeBracketsCount++;
+                }
+            }
+            difference = closeBracketsCount - openBracketsCount;
+            return difference;
+        }
+        
+        // method which checkcorrectness of record numbers in exponention form
+        private bool CheckRecordNumbersInExponentForm(string inputString)
```
Comment (IgorKheidorov) [inline Task_DEV-2/Checker.cs]: Use standard TryParse C# validators instead of this

File: Task_DEV-2/Converter.cs [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+
+        // concstructor 
+        public Converter(string inputString)
+        {
+            this.inputString = inputString;
+        }
+
+        // splittig string to array of string
+        // in this array elements is operator or variable
+        private string[] SplittingString()
```
Comment (IgorKheidorov) [inline Task_DEV-2/Converter.cs]: Can't understand implementation logics, try to decompose and simplify, add comments

File: Task_DEV-2/Calculator.cs (hunk 4 above)
Comment (VladislavZablocki) [reply Task_DEV-2/Calculator.cs:75]: i use conversion infix to postfix for ease of calculation
in postfix form we can go through the array from start to finish and do all operation in a necessary sequence
in infix form first we look for the priority of operation

**Pull request merged**